### PR TITLE
Add ability to control escaping for paths and queries

### PR DESCRIPTION
**Pull request by pakrym**
Re-enable some tests.

requires https://github.com/Azure/autorest.test-server/pull/18

### CODE REVIEW

File: test/AutoRest.TestServer.Tests/url/Generated/Operations/PathsOperations.cs [hunk 1]
```diff
@@ -45,7 +45,7 @@ public static async ValueTask<Response> GetBooleanFalseAsync(ClientDiagnostics c
                 request.Method = RequestMethod.Get;
                 request.Uri.Reset(new Uri($"{host}"));
                 request.Uri.AppendPath("/paths/bool/false/", false);
-                request.Uri.AppendPath("false", false);
+                request.Uri.AppendPath("false", true);
```
Comment (pakrym) [inline test/AutoRest.TestServer.Tests/url/Generated/Operations/PathsOperations.cs]: Eventually we would be doing escaping for constants as part of generation not yet tho.

File: src/AutoRest.CSharp.V3/Plugins/Modeler.cs [hunk 2]
```diff
@@ -190,6 +190,22 @@ private ConstantOrParameter[] ToParts(string httpRequestUri, Dictionary<string,
...
+        {
+            List<PathSegment> host = new List<PathSegment>();
+            foreach ((string text, bool isLiteral) in GetPathParts(httpRequestUri))
+            {
+                // WORKAROUND FOR https://github.com/Azure/autorest.modelerfour/issues/58
+                if (!parameters.ContainsKey(text))
+                {
+                    parameters[text] = StringConstant(text);
+                }
+                host.Add(new PathSegment(isLiteral ? StringConstant(text) : parameters[text], !isLiteral));
```
Comment (MiYanni) [inline src/AutoRest.CSharp.V3/Plugins/Modeler.cs]: This is duplicate code of the method above it with 1 line difference.

Comment (pakrym) [reply src/AutoRest.CSharp.V3/Plugins/Modeler.cs]: Cleaned up.

**Comment by pakrym** [comment]
> Is this PR for #242?

There are more features around escaping we need to implement (x-ms-skipescaping-...) but this is a start.

**Comment by pakrym** [comment]
I'll take this after the TestServer v1 PR goes in to verify I didn't break any tests.

**Pull request merged**